src/groupByKey.ts

```diff
@@ -2,6 +2,20 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(
+  items: object[], key: keyof object,
+): GroupsMap<object> {
+  const result: GroupsMap<object> = {};
+
+  items.forEach((item: object) => {
+    const keyValue: any = item[key];
```

Comment:
Try to avoid using any as a type

```suggestion
    const keyValue = item[key] as string;
```